prow/apis/prowjobs/v1/types.go

```diff
@@ -179,6 +179,10 @@ type ProwJobSpec struct {
 	// Presubmits and Postsubmits can also be set to hidden by
 	// adding their repository in Decks `hidden_repo` setting.
 	Hidden bool `json:"hidden,omitempty"`
+
+	// ProwJobDefault holds configuration options provided as defaults
+	// in the Prow config
+	ProwJobDefault *ProwJobDefault `json:"prowjob_defaults,omitempty"`
```

Comment:
Nit: I find this name confusing, why not directly have a `tenant_id` field? We can use a string pointer to differentiate empty string from unset

Reply:
The point of not having a tenant_id field and instead using this default configuration is so that it will be easy to set tenantIDs based on cluster or org/repo the same way we do for defaultDecorationConfig. This design was based on default decoration config. Open to suggestions about variable naming.

Reply:
It is also extensible for other ProwJob fields we may want to include in the future as we can do with the default decoration configs.

Reply:
Can we then maybe embedd and inline it?

It also seems to be a bit of a premature optimization for me. It  will only be useful if we have other config options that we want to set based on the exact same matching logic as `tenant_id`.

Reply:
Embedding was avoided to differentiate matching fields from configuration.

The majority of prow configuration defaulting logic is currently handled by org/repo indexing or cluster indexing so it seems likely to me that new fields would be satisfied by the existing matching fields. This is easily extensible to support more matching fields if needed.